move-vm-backend/src/lib.rs

```diff
@@ -177,6 +182,11 @@ where
     ) -> VmResult {
         let mut gas_handler = GasHandler::new(gas);
 
+        // MoveVM by default doesn't charge gas for publishing, so we need to do it manually here.
+        if let Err(result) = gas_handler.charge_publishing_to_storage(bundle.len()) {
```

Comment:
Let's move it after unpacking the bundle.

Reply:
Done